### DISCUSSION TITLE
etcd binding with ipv6

### DISCUSSION THREAD

**Discussion by Tochemey** [etcd-io/etcd#15982] [General]
I would like to know whether etcd can bind to IPv6.

For instance it is unable to bind with this IPv6 `[fe80::1]` which is valid.

**Comment by jmhbnz** [comment]
Hey @Tochemey - Thanks for your question, I've run etcd with ipv6 before using flags that look something like examples below:

```bash
 ./bin/etcd --listen-client-urls 'http://[::1]:2379' --advertise-client-urls 'http://[::1]:2379'
 ```
 
Can you share the example configuration flags or file you are using that isn't working as you expect?

**Reply by Tochemey** [reply]
@jmhbnz I am using the embed server. However this is the configuration I was starting with:
```
{"level":"info","ts":"2023-05-30T09:58:20.842062Z","caller":"embed/etcd.go:127","msg":"configuring peer listeners","listen-peer-urls":["http://192.168.8.107:2380","http://[fe80::109d:622f:e150:7ade]:2380","http://[fe80::1]:2380","http://[fe80::4019:bb5e:bf0:1813]:2380","http://[fe80::61fc:3b73:fdd4:1ab3]:2380","http://[fe80::7c77:424e:e5c0:3225]:2380","http://[fe80::aede:48ff:fe00:1122]:2380","http://[fe80::b09c:4aff:feba:701e]:2380","http://[fe80::b0f9:e0ff:fe6d:aae1]:2380","http://[fe80::ce81:b1c:bd2c:69e]:2380","http://[fe80::ed1b:e9be:8158:5397]:2380"]}
```
Those addresses are my host addresses. This is the error message I got:
```
"msg":"failed to start the etcd embedded server: listen tcp [fe80::109d:622f:e150:7ade]:2380: bind: can't assign requested address"}
```

**Reply by Tochemey** [reply]
@jmhbnz Thanks for the assistance. I think most of those addresses are link-local
